packages/next/server/render.tsx

```diff
@@ -551,6 +551,18 @@ export async function renderToHTML(
 
   const pageIsDynamic = isDynamicRoute(pathname)
 
+  if (
+    renderOpts.nextExport &&
+    (hasPageGetInitialProps || defaultAppGetInitialProps)
+  ) {
```

Comment:
```suggestion
  if (renderOpts.nextExport && hasPageGetInitialProps) {
```
The `defaultAppGetInitialProps` value here signals if the user hasn't provided a custom `getInitialProps` method in `_app` so I think we want to omit this from the check.